File: sample/src/main/java/com/qingmei2/sample/system/SystemActivity.java
package com.qingmei2.sample.system;

import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.ImageView;

import com.qingmei2.rximagepicker.core.RxImagePicker;
import com.qingmei2.rximagepicker.entity.Result;
import com.qingmei2.rximagepicker.ui.SystemImagePicker;
import com.qingmei2.sample.R;
import com.qingmei2.sample.imageloader.GlideApp;

import io.reactivex.functions.Consumer;

@SuppressWarnings("CheckResult")
public class SystemActivity extends AppCompatActivity {

    private SystemImagePicker defaultImagePicker;
    private ImageView ivPickedImage;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_system);

        ivPickedImage = findViewById(R.id.imageView);
        FloatingActionButton fabCamera = findViewById(R.id.fabPickCamera);
        FloatingActionButton fabGallery = findViewById(R.id.fabGallery);

        initRxImagePicker();
        fabCamera.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                pickCamera();
            }
        });
        fabGallery.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                pickGallery();
            }
        });
    }

    private void initRxImagePicker() {
        defaultImagePicker = RxImagePicker.INSTANCE.create();
    }

    private void pickGallery() {
        defaultImagePicker.openGallery()
                .subscribe(new Consumer<Result>() {
                    @Override
                    public void accept(Result result) throws Exception {
                        onPickUriSuccess(result.getUri());
                    }
                });
    }

    private void pickCamera() {
        defaultImagePicker.openCamera()
                .subscribe(new Consumer<Result>() {
                    @Override
                    public void accept(Result result) throws Exception {
                        onPickUriSuccess(result.getUri());
                    }
                });
    }

    private void onPickUriSuccess(Uri uri) {
        GlideApp.with(this)
                .load(uri)
                .into(ivPickedImage);
    }
}

File: sample/src/main/java/com/qingmei2/sample/system/SystemActivity.kt
package com.qingmei2.sample.system

import android.annotation.SuppressLint
import android.net.Uri
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.qingmei2.rximagepicker.core.RxImagePicker
import com.qingmei2.rximagepicker.ui.gallery.DefaultGalleryMimes
import com.qingmei2.rximagepicker.ui.gallery.DefaultSystemGalleryConfig
import com.qingmei2.sample.R
import com.qingmei2.sample.imageloader.GlideApp
import kotlinx.android.synthetic.main.activity_system.*

@SuppressLint("CheckResult")
class SystemActivity : AppCompatActivity() {

    private lateinit var defaultImagePicker: SystemImagePicker

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_system)

        initRxImagePicker()
        fabPickCamera.setOnClickListener { pickCamera() }
        fabGallery.setOnClickListener { pickGallery() }
    }

    private fun initRxImagePicker() {
        defaultImagePicker = RxImagePicker.create(SystemImagePicker::class.java)
    }

    private fun pickGallery() {
        defaultImagePicker
                .openGallery(
                        this,
                        DefaultSystemGalleryConfig.instance(
                                // mimesType = DefaultGalleryMimes.videoOnly()     // only video files
                                // mimesType = DefaultGalleryMimes.imageOnly()     // only image files, default options.
                                // mimesType = DefaultGalleryMimes.audioOnly()     // only audio files
                                mimesType = DefaultGalleryMimes.customTypes("video/*;image/*") // multiType
                        )
                )
                .subscribe { result -> onPickUriSuccess(result.uri) }
    }

    private fun pickCamera() {
        defaultImagePicker.openCamera(this)
                .subscribe { result -> onPickUriSuccess(result.uri) }
    }

    private fun onPickUriSuccess(uri: Uri) {
        GlideApp.with(this)
                .load(uri)
                .into(imageView)
    }
}
